pydoc/developer/BENCHMARK.md

```diff
@@ -0,0 +1,504 @@
+# PyTest Benchmarks
+
+Arkouda uses `pytest-benchmark` for performance benchmarking. This document provides an overview of running `pytest-benchmark` and the configurations available to the user.
+
+More information on `pytest-benchmark` can be found [here](https://pytest-benchmark.readthedocs.io/en/latest/)
+
+## Running The Full Suite
+In most cases, running the full benchmark suite is desired. The simplest way to do this is to navigate into the 
+root-level of arkouda and run `make benchmark`
+
+This will run the entire benchmark suite with the following command:
+```commandline
+python3 -m pytest -c benchmark.ini --benchmark-autosave --benchmark-storage=file://benchmark_v2/.benchmarks
+```
+
+The results for the benchmarks can be found within the provided benchmark storage path, which by default is within a
+directory found in `//benchmark_v2/.benchmarks`.
+
+The command used here will provide a full set of information on all benchmarks available. 
+
+The `-c` flag specifies to PyTest to use `benchmark.ini` as the configuration file for this set of test. The 
+configuration file specifies which files contain benchmarks as well as a set of environment variables used by 
+the benchmarks.
+
+`--benchmark-autosave` tells `pytest-benchmark` to save the results of the benchmark in a json file stored in the
+path specified by `--benchmark-storage`.

```

Comment:
I would leave these here, but also I was hoping to add `-c` and `-k` to the `Benchmark Arguments` Section.

Reply:
Added -c and -k to the arguments section